tools/SetupFlow/DevHome.SetupFlow/ViewModels/Environments/SelectEnvironmentProviderViewModel.cs

```diff
@@ -0,0 +1,104 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.ObjectModel;
+using System.Threading.Tasks;
+using CommunityToolkit.Mvvm.ComponentModel;
+using CommunityToolkit.Mvvm.Input;
+using CommunityToolkit.Mvvm.Messaging;
+using DevHome.Common.Contracts.Services;
+using DevHome.Common.Environments.Models;
+using DevHome.SetupFlow.Models.Environments;
+using DevHome.SetupFlow.Services;
+using Serilog;
+
+namespace DevHome.SetupFlow.ViewModels.Environments;
+
+public partial class SelectEnvironmentProviderViewModel : SetupPageViewModelBase
+{
+    private readonly ILogger _log = Log.ForContext("SourceContext", nameof(SelectEnvironmentProviderViewModel));
+
+    private readonly Microsoft.UI.Dispatching.DispatcherQueue _dispatcher;
+
+    private readonly IComputeSystemService _computeSystemService;
+
+    public ComputeSystemProviderDetails SelectedProvider { get; private set; }
+
+    [ObservableProperty]
+    private bool _areProvidersLoaded;
+
+    [ObservableProperty]
+    private int _selectedProviderIndex;
+
+    [ObservableProperty]
+    private ObservableCollection<ComputeSystemProviderViewModel> _providersViewModels;
+
+    public SelectEnvironmentProviderViewModel(
+        ISetupFlowStringResource stringResource,
+        SetupFlowOrchestrator orchestrator,
+        IComputeSystemService computeSystemService)
+           : base(stringResource, orchestrator)
+    {
+        PageTitle = stringResource.GetLocalized(StringResourceKey.SelectEnvironmentPageTitle);
+        _computeSystemService = computeSystemService;
+        _dispatcher = Microsoft.UI.Dispatching.DispatcherQueue.GetForCurrentThread();
+    }
+
+    public void LoadProviders()
+    {
+        AreProvidersLoaded = false;
+        Orchestrator.NotifyNavigationCanExecuteChanged();
+
+        Task.Run(async () =>
```

Comment:
`await`

Reply:
I wanted this to fire and not need to await the call, but if you feel strongly about it I can add it.

Reply:
added